### PR TITLE
Get ready for pipeline-jobs

### PR DESCRIPTION
**Pull request by neuhausjulian**
Replaced AbstractProject.class with Job.class to support also Pipeline jobs.

Upgraded jenkins-version in pom.xml

Fixed some reports from FindBug.

### CODE REVIEW

**Comment by BrendanBall** [comment]
@neuhausjulian I'm not sure if the maintainer is still active but I would really like this for pipeline jobs as well. Maybe we can create a permanent fork of this plugin?

**Comment by neuhausjulian** [comment]
@BrendanBall Actually I have time to wait. Yes, it looks like there is no maintainer left but as I mentioned I can wait. (we installed the plugin manually ) If you need my changes you can fork my fork already including the fix. 
If you start your own fork I'm glad to participate.

**Comment by BrendanBall** [comment]
@neuhausjulian I haven't bothered forking yet, I've just downloaded and built your fork. The plugin now allows me to specify a time for the build, previously it couldn't even load the page. However it still doesn't work properly. When I try schedule a build with the plugin it says in the logs "no valid crumb was included in request. Returning 403". This is apparently a problem with moving from Jenkins 1 to Jenkins 2. Which version are you using? I haven't figured out how to fix it yet.

**Comment by neuhausjulian** [comment]
@BrendanBall we also have the problem with the invalid crumb. This happens because the plugin makes a post-ajax-request with out a crumb. So if you have CSRF-Protection actviated you will have to add a valid crumb to the request. I added a lazy workaround to our local branch. We only have projects with additional build-params, so we just needed a simple redirect via GET to "https://YOUR_URL/job/PROJECT/build" where we can set the params and trigger the delayed build.

So I just replaced the Ajax-POST-Request which triggers the build with a redirect. If we had also a project without params, we would have the same problem.

Long story short: Yeah there is a problem and it has to be fixed in a better way as I did :D

**Comment by neuhausjulian** [comment]
maybe I can fix this in my fork. I will also add these changes to this pull-request. I'm not really sure how to add my fork as possible Plugin into the jenkins-store.

**Comment by BrendanBall** [comment]
@neuhausjulian I guess you'll have to name it something different

**Comment by pingunaut** [comment]
@neuhausjulian I created a pull request to your fork fixing the CSRF-Protection issue.
Would be great if you merge it and hopefully we can get #6 merged into upstream.
What's the best thing to do if @anderl86 does no longer maintain the plugin? Personally I think it's better to find have new maintainers instead of creating a fork (would be glad to help out btw). I'll try to get something going in the mailing list...

**Comment by pingunaut** [comment]
@neuhausjulian I'd like to merge this pull request, but there is still the open topic regarding csrf tokens. did you have the chance to take a look at https://github.com/neuhausjulian/schedule-build-plugin/pull/1 in the meantime?
If not no problem, I would then suggest, that I open another pull request to add csrf support afterwards. 
Think, having smaller pull requests is a good idea to keep changes comprehensable..
Any objections?
